astropy/coordinates/name_resolve.py

```diff
@@ -132,34 +132,30 @@ def get_icrs_coordinates(name):
             fmt_url = fmt_url.format(name=urllib.parse.quote(name), db=db)
             urls.append(fmt_url)
 
+    exceptions = []
     for url in urls:
         try:
             # Retrieve ascii name resolve data from CDS
             resp = urllib.request.urlopen(url, timeout=data.conf.remote_timeout)
             resp_data = resp.read()
             break
         except urllib.error.URLError as e:
-            # This catches a timeout error, see:
-            #   http://stackoverflow.com/questions/2712524/handling-urllib2s-timeout-python
-            if isinstance(e.reason, socket.timeout):
-                # If it was a timeout, try with the next URL
-                continue
-            else:
-                raise NameResolveError(
-                    "Unable to retrieve coordinates for name '{0}'; "
-                    "connection timed out".format(name))
-        except socket.timeout:
+            exceptions.append(e)
+            continue
+        except socket.timeout as e:
             # There are some cases where urllib2 does not catch socket.timeout
             # especially while receiving response data on an already previously
             # working request
-            raise NameResolveError(
-                "Unable to retrieve coordinates for name '{0}'; connection "
-                "timed out".format(name))
+            exceptions.append(e)
+            continue
 
-    # All Sesame URL's timed out...
+    # All Sesame URL's failed...
     else:
-        raise NameResolveError("All Sesame queries timed out. Unable to "
-                               "retrieve coordinates.")
+        messages = ["{url}: {e.reason}".format(url=url, e=e)
```

Comment:
Is this backward compatible? I don't think so. Astropy still needs to support Python 2 for now.

1. Change string formatting back to `"{}"` style.
2. Use `str(e)` instead of `.reason`.